src/metabase/sync/analyze/fingerprint.clj

```diff
@@ -135,23 +135,30 @@
          [:< :fingerprint_version version]
          [:in :base_type not-yet-seen]]))))
 
+(def ^:private base-clause
+  [:and
+   [:= :active true]
+   [:or
+    [:not (mdb/isa :special_type :type/PK)]
+    [:= :special_type nil]]
+   [:not-in :visibility_type ["retired" "sensitive"]]
+   [:not= :base_type "type/Structured"]])
+
+(def ^:dynamic *refingerprint?*
+  "Whether we are refingerprinting or doing the normal fingerprinting. Refingerprinting should get fields that already
+  are analyzed and have fingerprints."
+  false)
+
 (s/defn ^:private honeysql-for-fields-that-need-fingerprint-updating :- {:where s/Any}
   "Return appropriate WHERE clause for all the Fields whose Fingerprint needs to be re-calculated."
   ([]
-   {:where [:and
-            [:= :active true]
-            [:or
-             [:not (mdb/isa :special_type :type/PK)]
-             [:= :special_type nil]]
-            [:not-in :visibility_type ["retired" "sensitive"]]
-            [:not= :base_type "type/Structured"]
-            (cons :or (versions-clauses))]})
+   {:where (cond-> base-clause
+             (not *refingerprint?*) (conj (cons :or (versions-clauses))))})
```

Comment:
this was a carryover from previous code. Seems 6 vs half dozen to me.

Reply:
The reason I brought it up is because there are a lot of little Clojure gotchas you have to go through: cons creates a list which means that conj prepends rather than appends and all this in a threading macro so the arg positions aren't immediately glanceable. But no biggie

Reply:
i think i'll leave it like this because it would actually need to be `(apply vector :or (versions-clauses))` whereas the cons expects an element and a seq and hooks them together, which is what we have. I agree that things get a bit murky but the `conj` here is appending to the threaded `base-clause` (now renamed) which is a vector.